### PR TITLE
Add appmap-agent-init command

### PR DESCRIPTION
**Pull request by apotterri**

### CODE REVIEW

File: appmap/_implementation/configuration.py [hunk 1]
```diff
@@ -25,19 +26,75 @@ def warn_config_missing(path):
...
+def default_app_name(rootdir):
+    rootdir = Path(rootdir)
+    if not (rootdir / '.git').exists():
+        return rootdir.name
+
+    git = utils.git(cwd=str(rootdir))
+    repo_root = git('rev-parse --show-toplevel')
+    return Path(repo_root).name
+
+def find_top_packages(rootdir):
```
Comment (torchello) [inline appmap/_implementation/configuration.py:39]: I like the idea. I plan to implement something similar in Ruby as well.

File: appmap/_implementation/configuration.py [hunk 2]
```diff
@@ -72,21 +129,46 @@ def name(self):
...
+            'packages': [{'path': p} for p in find_top_packages(root_dir)],
+        }
+
     @property
     @lru_cache(maxsize=None)
     def _config(self):
-        path = Path(Env.current.get("APPMAP_CONFIG", "appmap.yml")).resolve()
+        # Only use a default config if the user hasn't specified a
+        # config.
+        env_config = Env.current.get("APPMAP_CONFIG")
```
Comment (torchello) [inline appmap/_implementation/configuration.py:146]: Do we need to support `APPMAP_CONFIG` env var for code extensions? Feels like the initial flow would not ask user to set env vars.

Comment (torchello) [reply appmap/_implementation/configuration.py:146]: Ruby CLI supports passing `-c,` `--config` params to set custom path though. I think we could focus on using the default path in the first phase.

**Review by torchello — APPROVED** [review]
Left comment regarding the usage of env vars.

**Comment by kgilpin** [comment]
APPMAP_CONFIG is not required

File: appmap/_implementation/configuration.py (hunk 2 above)
Comment (apotterri) [reply appmap/_implementation/configuration.py:146]: `APPMAP_CONFIG` only needs to be used at runtime, the CLI doesn't need to support it (as Kevin said).

We can add support for `--config` to `appmap-agent-init` when the code extensions need it.

**Pull request merged**